test_common/harness/imageHelpers.cpp

```diff
@@ -3671,45 +3671,201 @@ bool find_format( cl_image_format *formatList, unsigned int numFormats, cl_image
+					bsRGBA = true;
+				}
+			}
+			else
+			{
+				uiTotalFormatSize += sizeof(fullProfSRGBFormats);
+				bsRGBA = true;
+			}
+		}
+	}
+
+	cl_image_format *readFormatsToSupport = (cl_image_format*)malloc(uiTotalFormatSize);
+	cl_image_format *writeFormatsToSupport = (cl_image_format*)malloc(uiTotalFormatSize);
+
+	if (readFormatsToSupport == NULL || writeFormatsToSupport == NULL)
+	{
+		passed = false;
+		goto exit;
+	}
+
+	// Copy all required formats for the device profile and image type
+	if (gIsEmbedded)
+	{
+		memcpy(readFormatsToSupport,
+		       embeddedProfReadOrWriteFormats,
+		       sizeof(embeddedProfReadOrWriteFormats));
+
+		memcpy(writeFormatsToSupport,
+		       embeddedProfReadOrWriteFormats,
+		       sizeof(embeddedProfReadOrWriteFormats));
+	}
+	else
+	{
+		size_t uiReadOffset = 0;
+		size_t uiWriteOffset = 0;
+
+		memcpy(readFormatsToSupport + uiReadOffset,
+		       fullProfReadOrWriteFormats,
+		       sizeof(fullProfReadOrWriteFormats));
+		uiReadOffset += sizeof(fullProfReadOrWriteFormats) / sizeof(fullProfReadOrWriteFormats[0]);
+
+		memcpy(writeFormatsToSupport + uiWriteOffset,
+		       fullProfReadOrWriteFormats,
+		       sizeof(fullProfReadOrWriteFormats));
+		uiWriteOffset += sizeof(fullProfReadOrWriteFormats) / sizeof(fullProfReadOrWriteFormats[0]);
+
+		if (bDepth)
+		{
+			memcpy(readFormatsToSupport + uiReadOffset,
+			       fullProfReadOrWriteDepthFormats,
+			       sizeof(fullProfReadOrWriteDepthFormats));
+			uiReadOffset += sizeof(fullProfReadOrWriteDepthFormats) / sizeof(fullProfReadOrWriteDepthFormats[0]);
+
+			memcpy(writeFormatsToSupport + uiWriteOffset,
+			       fullProfReadOrWriteDepthFormats,
+			       sizeof(fullProfReadOrWriteDepthFormats));
+			uiWriteOffset += sizeof(fullProfReadOrWriteDepthFormats) / sizeof(fullProfReadOrWriteDepthFormats[0]);
+		}
+
+		if (bsRGBA)
+		{
+			memcpy(readFormatsToSupport + uiReadOffset,
+			       fullProfSRGBFormats,
+			       sizeof(fullProfSRGBFormats));
+			uiReadOffset += sizeof(fullProfSRGBFormats) / sizeof(fullProfSRGBFormats[0]);
+
+			memcpy(writeFormatsToSupport + uiWriteOffset,
+			       fullProfSRGBFormats,
+			       sizeof(fullProfSRGBFormats));
+			uiWriteOffset += sizeof(fullProfSRGBFormats) / sizeof(fullProfSRGBFormats[0]);
+		}
+	}
 
     cl_image_format *formatsToTest;
     unsigned int testCount;
-    bool passed = true;
 
+	// Work out the number of elements in the format array
     if( flags == CL_MEM_READ_ONLY )
     {
         formatsToTest = readFormatsToSupport;
-        testCount = sizeof( readFormatsToSupport ) / sizeof( readFormatsToSupport[ 0 ] );
+        testCount = uiTotalFormatSize / sizeof( readFormatsToSupport[ 0 ] );
     }
     else
     {
         formatsToTest = writeFormatsToSupport;
-        testCount = sizeof( writeFormatsToSupport ) / sizeof( writeFormatsToSupport[ 0 ] );
+        testCount = uiTotalFormatSize / sizeof( writeFormatsToSupport[ 0 ] );
     }
 
     for( unsigned int i = 0; i < testCount; i++ )
```

Comment:
The message below needs to be updated with the actual version supported by the device.